### PR TITLE
Initial working version for Factorio 2.0 Space Age

### PR DESCRIPTION
**Pull request by MaciejGorczyca**
Hey.

I had no idea what changed etc but I was able to feed Claude 3.5 Sonnet with "Modding" changes from [change log for 2.0.7](https://wiki.factorio.com/Version_history/2.0.0) plus errors I was getting from Factorio and it successfully spit out correct fixes for like 95% of the things.

Checked if everything works in editor and it does. The only issue is that gunship, jet and flying fortress seem to appear in recipes menu even when they shouldn't. Surprisingly cargo plane does NOT but there is no difference in code between other airplanes and cargo plane...

The TL;DR Mod is in a working state but it could get some code polish if anyone is willing to pick it up.

### CODE REVIEW

**Comment by MaciejGorczyca** [comment]
Aircraft 2.0.0 attached below if anyone wants to manually install mod and play with it.

[Aircraft_2.0.0.zip](https://github.com/user-attachments/files/17517419/Aircraft_2.0.0.zip)

**Comment by nicholasgower** [comment]
Thank you for making this fork of Aircraft. I have built on your work to make a version of Aircraft that removes the recipe bug you noted, and that also integrates features specific to 2.0 and Space Age. I consider this fully playable.

---------------------------------------------------------------------------------------------------
Version: 2.0.1
Date: 29-10-2024
  Changes:
    - Fixed remaining bugs noted by MaciejGorczyca.
    - All aircraft now support logistic requests and can be remotely flown.
    - Changed "Advanced Aerodynamics" technology to unlock after crafting 100 robot frames.
    - Changed "Aircraft Energy Shield" technology to require researching gunships.
  Changes when Space Age is enabled:
    - Aircraft can only be placed on Nauvis and Fulgora. Aircraft require at least 700 hPa pressure and at most 15 m/s^2 gravity.
    - "Jets" technology requires Space and Agricultural science packs. Recipe changed to require Carbon Fiber.
    - "Flying Fortress" technology requires Metallurgic and Agricultural science packs. Recipe changed to require Carbon Fiber.
    - "Aircraft Energy Shield" technology requires Space and Electromagnetic science packs.
    - "Afterburner" technology requires Space science packs.
    - "Rocket Silo" technology requires "Cargo Planes" as a prerequisite.
    - "Space platform thruster" technology requires "Afterburner" as a prerequisite.
---------------------------------------------------------------------------------------------------
Version: 2.0.0
Date: 24-10-2024
  Changes:
    - Updated for 2.0 (Thanks MaciejGorczyca for feeding original code to Claude 3.5 Sonnet to fix most compatibility issues)
---------------------------------------------------------------------------------------------------

[Aircraft_2.0.1.zip](https://github.com/user-attachments/files/17564371/Aircraft_2.0.1.zip)

**Comment by duchenpaul** [comment]
Nice job done, thanks, I will find time to test it out!

File: prototypes/entities.lua [hunk 1]
```diff
@@ -117,7 +117,10 @@ local function add_recurrent_params(craft)
   craft.dying_explosion = "medium-explosion"
   craft.terrain_friction_modifier = 0
   craft.collision_box = {{-0.9, -1.3}, {0.9, 1.3}}
-  craft.collision_mask = {}
+  -- Original
```
Comment (duchenpaul) [inline prototypes/entities.lua:120]: We should remove the "added due to 2.0 modding API changes" comments, as the git commit has indicated the changes.

File: prototypes/items.lua [hunk 2]
```diff
@@ -270,6 +272,8 @@ data:extend({
   { -- Napalm
     type = "ammo",
     name = "napalm",
+    -- added due to 2.0 modding API changes
```
Comment (duchenpaul) [inline prototypes/items.lua:275]: We should remove the "added due to 2.0 modding API changes" comments, as the git commit has indicated the changes.

File: prototypes/items.lua [hunk 3]
```diff
@@ -240,6 +240,8 @@ data:extend({
   { -- High explosive cannon shell
     type = "ammo",
     name = "high-explosive-cannon-shell",
+    -- added due to 2.0 modding API changes
```
Comment (duchenpaul) [inline prototypes/items.lua:243]: We should remove the "added due to 2.0 modding API changes" comments, as the git commit has indicated the changes.

File: prototypes/particles.lua [hunk 4]
```diff
@@ -29,7 +29,12 @@ napalm_particle.tint = napalm_tint
 
 local napalm_fire_flame = table.deepcopy(data.raw["fire"]["fire-flame"])
 napalm_fire_flame.name = "napalm_fire_flame"
-napalm_fire_flame.emissions_per_second = 0.050
+--napalm_fire_flame.emissions_per_second = 0.050
+--change in 2.0 modding API
```
Comment (duchenpaul) [inline prototypes/particles.lua:33]: We should remove the "added due to 2.0 modding API changes" comments, as the git commit has indicated the changes.

**Comment by MaciejGorczyca** [comment]
Great stuff @nicholasgower !

There is one thing I would like to mention:

>  - Aircraft can only be placed on Nauvis and Fulgora. Aircraft require at least 700 hPa pressure and at most 15 m/s^2 gravity.

It would be awesome if there was a toggle in settings where you can make airplanes work where they make sense like you did above OR on every surface like it does now, so players can pick if they want realistic or easy/functional way. The default could be set to realistic.

Personally I use airplanes so I don't get stuck on random entities in my way or cliffs/water etc.

PS: I don't really want to get involved further with coding/testing etc so feel free to change any comments/features. I just wanted to get the Jet working. :)

PS2: If the original author of the mod is not active anymore maybe someone could pick it up, fork, pull all the changes and submit the mod under new page/name on Factorio modding website? The license is MIT so we can do pretty much whatever we want with the code.

**Comment by nerfkidjcb** [comment]
> Great stuff @nicholasgower !
> 
> There is one thing I would like to mention:
> 
> > * Aircraft can only be placed on Nauvis and Fulgora. Aircraft require at least 700 hPa pressure and at most 15 m/s^2 gravity.
> 
> It would be awesome if there was a toggle in settings where you can make airplanes work where they make sense like you did above OR on every surface like it does now, so players can pick if they want realistic or easy/functional way. The default could be set to realistic.
> 
> Personally I use airplanes so I don't get stuck on random entities in my way or cliffs/water etc.
> 
> PS: I don't really want to get involved further with coding/testing etc so feel free to change any comments/features. I just wanted to get the Jet working. :)
> 
> PS2: If the original author of the mod is not active anymore maybe someone could pick it up, fork, pull all the changes and submit the mod under new page/name on Factorio modding website? The license is MIT so we can do pretty much whatever we want with the code.

If this doesn't get picked up in the next week by someone I'll do it :)

Edit: looks like someone did (but closed source grrr)  https://mods.factorio.com/mod/Aircraft-space-age

**Comment by nicholasgower** [comment]
> > Great stuff @nicholasgower !
> > There is one thing I would like to mention:
> > > * Aircraft can only be placed on Nauvis and Fulgora. Aircraft require at least 700 hPa pressure and at most 15 m/s^2 gravity.
> > 
> > 
> > It would be awesome if there was a toggle in settings where you can make airplanes work where they make sense like you did above OR on every surface like it does now, so players can pick if they want realistic or easy/functional way. The default could be set to realistic.
> > Personally I use airplanes so I don't get stuck on random entities in my way or cliffs/water etc.
> > PS: I don't really want to get involved further with coding/testing etc so feel free to change any comments/features. I just wanted to get the Jet working. :)
> > PS2: If the original author of the mod is not active anymore maybe someone could pick it up, fork, pull all the changes and submit the mod under new page/name on Factorio modding website? The license is MIT so we can do pretty much whatever we want with the code.
> 
> If this doesn't get picked up in the next week by someone I'll do it :)
> 
> Edit: looks like someone did (but closed source grrr) https://mods.factorio.com/mod/Aircraft-space-age

I am MeteorSwarm. Sorry for the confusion. Here is the code repo for this mod. You're welcome to modify this for your own ends. However, just because the code doesn't have a repo doesn't mean the code is closed source. Like the original Aircraft, my fork is under the MIT license, meaning anyone can download the compressed mod, decompress it, and modify the code. https://github.com/nicholasgower/Aircraft-Space-Age/tree/master

**Comment by nicholasgower** [comment]
Also, I think that adding a setting that allows aircraft on all planets is a good idea for now, but a better long-term goal is expanding the Aircraft mod to introduce different types of flying vehicles that work under more extreme conditions. I'll add an option to unrestrict aircraft, but that should not be the end.

After reviewing the gravity and pressure of Gleba, I now believe that aircraft that work on Nauvis should also work on Gleba. While Gleba has twice the gravity of Nauvis, it has twice the pressure, so they are similiar enough that the same aircraft should work. That leaves only Vulcanus and Aquilo as the only planets that custom aircraft make sense for.

For these new aircraft, I have the following ideas:

Blimp: Flies via buoyancy without concern for gravity. Low energy usage, but low speed and low HP. Intended for use on Vulcanus, but can be used on any planet with an atmosphere.

Rocket Plane: Works on low-pressure planets. Very high energy usage and high speed. Only works on Aquilo.

I believe these changes would give Vulcanus and Aquilo a different feel for how aircraft work without taking the option away. This is a longterm goal, so for now, I will add an option to unrestrict aircraft based on surface.
